homeassistant/util/dt.py

```diff
@@ -118,6 +117,8 @@ def parse_datetime(dt_str: str) -> dt.datetime:
     if kws['microsecond']:
         kws['microsecond'] = kws['microsecond'].ljust(6, '0')
     tzinfo_str = kws.pop('tzinfo')
+
+    tzinfo = None  # type: Optional[dt.tzinfo]
```

Comment:
I would hold on a bit until mypy greens to remove the in-code checks. (Since the python interpreter currently does not check typing)